cairo/ethereum/crypto/alt_bn128.cairo

```diff
@@ -438,6 +446,309 @@ func bnf12_add{
+
+    // Continue with next term
+    return compute_polynomial_product(a, b, prime, i, j + 1);
+}
+
+func create_bnf12_from_dict{range_check_ptr, mul_dict: DictAccess*}() -> BNF12 {
+    alloc_locals;
+
+    let (result_struct: BNF12Struct*) = alloc();
+
+    let (c0_ptr) = dict_read{dict_ptr=mul_dict}(0);
+    let (c1_ptr) = dict_read{dict_ptr=mul_dict}(1);
+    let (c2_ptr) = dict_read{dict_ptr=mul_dict}(2);
+    let (c3_ptr) = dict_read{dict_ptr=mul_dict}(3);
+    let (c4_ptr) = dict_read{dict_ptr=mul_dict}(4);
+    let (c5_ptr) = dict_read{dict_ptr=mul_dict}(5);
+    let (c6_ptr) = dict_read{dict_ptr=mul_dict}(6);
+    let (c7_ptr) = dict_read{dict_ptr=mul_dict}(7);
+    let (c8_ptr) = dict_read{dict_ptr=mul_dict}(8);
+    let (c9_ptr) = dict_read{dict_ptr=mul_dict}(9);
+    let (c10_ptr) = dict_read{dict_ptr=mul_dict}(10);
+    let (c11_ptr) = dict_read{dict_ptr=mul_dict}(11);
+
+    tempvar zero = new UInt384(0, 0, 0, 0);
+    let coeff_ptr = cast(c0_ptr, UInt384*);
+    let c0 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c0 = c0;
+
+    let coeff_ptr = cast(c1_ptr, UInt384*);
+    let c1 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c1 = c1;
+
+    let coeff_ptr = cast(c2_ptr, UInt384*);
+    let c2 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c2 = c2;
+
+    let coeff_ptr = cast(c3_ptr, UInt384*);
+    let c3 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c3 = c3;
+
+    let coeff_ptr = cast(c4_ptr, UInt384*);
+    let c4 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c4 = c4;
+
+    let coeff_ptr = cast(c5_ptr, UInt384*);
+    let c5 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c5 = c5;
+
+    let coeff_ptr = cast(c6_ptr, UInt384*);
+    let c6 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c6 = c6;
+
+    let coeff_ptr = cast(c7_ptr, UInt384*);
+    let c7 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c7 = c7;
+
+    let coeff_ptr = cast(c8_ptr, UInt384*);
+    let c8 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c8 = c8;
+
+    let coeff_ptr = cast(c9_ptr, UInt384*);
+    let c9 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c9 = c9;
+
+    let coeff_ptr = cast(c10_ptr, UInt384*);
+    let c10 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c10 = c10;
+
+    let coeff_ptr = cast(c11_ptr, UInt384*);
+    let c11 = _process_coefficient(coeff_ptr, zero);
+    assert result_struct.c11 = c11;
+
+    tempvar bnf12_result = BNF12(result_struct);
+    return bnf12_result;
+}
+
+func _process_coefficient(coeff_ptr: UInt384*, zero: UInt384*) -> UInt384 {
+    if (coeff_ptr == 0) {
+        tempvar coeff = [zero];
+    } else {
+        let coeff = [cast(coeff_ptr, UInt384*)];
+        tempvar coeff = coeff;
+    }
+    return coeff;
+}
+
+func _get_bnf12_coeff(a: BNF12, i: felt) -> UInt384 {
+    if (i == 0) {
```

Comment:
this should be a jump table (like for opcodes and precompiles)

Reply:
well actually it's even simpler, you're basically indexing a segment here. so you can just replace all of this with 

```
func _get_bnf12_coeff(a: BNF12, i: felt) -> UInt384 {
    let a_segment = cast(a.value, UInt384*);
    let coeff = a_segment[i];
    return coeff;
}
```

and you don't even need the function anymore, you can just inline this.